packages/core/src/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, h, Prop, Watch, Element } from "@stencil/core";
+import clsx from "clsx";
+import { tabbable } from "tabbable";
+import { v4 as uuidv4 } from "uuid";
+
+@Component({
+  tag: "dso-dropdown-menu",
+  shadow: true,
+})
+export class DropdownMenu {
+  /**
+   * Whether the menu is open or closed.
+   * This attribute is reflected and mutable.
+   */
+  @Prop({ reflect: true, mutable: true })
+  open = false;
+
+  /**
+   * Alignment of the dropdown
+   */
+  @Prop()
+  dropdownAlign: "left" | "right" = "left";
+
+  @Element()
+  el: HTMLElement | undefined;
+
+  get host(): HTMLElement {
+    if (!this.el) {
+      throw new ReferenceError("Host element not found");
+    }
+
+    return this.el;
+  }
+
+  get button(): HTMLButtonElement {
+    let button = this.host.getElementsByTagName("button")[0];
```

Comment:
`button` wijzigt niet meer dus kan een `const` worden. Daarnaast zou ik de selector uitbreiden met het slot argument: `button[slot="button"]`: Het staat de implementator vrij eerst de inhoud en dan de button te plaatsen. Als de inhoud een button bevat krijgen we raar gedrag.

Reply:
Aangepast